zebra-consensus/src/primitives/redpallas/tests.rs

```diff
@@ -0,0 +1,82 @@
+//! Tests for redpallas signature verification
+
+use super::*;
+
+use std::time::Duration;
+
+use color_eyre::eyre::{eyre, Result};
+use futures::stream::{FuturesUnordered, StreamExt};
+use tower::ServiceExt;
+use tower_batch::Batch;
+
+async fn sign_and_verify<V>(mut verifier: V, n: usize) -> Result<(), V::Error>
+where
+    V: Service<Item, Response = ()>,
+{
+    let mut rng = thread_rng();
+    let mut results = FuturesUnordered::new();
+    for i in 0..n {
+        let span = tracing::trace_span!("sig", i);
+        let msg = b"BatchVerifyTest";
+
+        match i % 2 {
+            0 => {
+                let sk = SigningKey::<SpendAuth>::new(&mut rng);
+                let vk = VerificationKey::from(&sk);
+                let sig = sk.sign(&mut rng, &msg[..]);
+                verifier.ready_and().await?;
+                results.push(span.in_scope(|| verifier.call((vk.into(), sig, msg).into())))
+            }
+            1 => {
+                let sk = SigningKey::<Binding>::new(&mut rng);
+                let vk = VerificationKey::from(&sk);
+                let sig = sk.sign(&mut rng, &msg[..]);
+                verifier.ready_and().await?;
+                results.push(span.in_scope(|| verifier.call((vk.into(), sig, msg).into())))
+            }
+            _ => panic!(),
+        }
+    }
+
+    while let Some(result) = results.next().await {
+        result?;
+    }
+
+    Ok(())
```

Comment:
Minor: just wanted to mention an alternative way to write this, even though I'm not really sure if it's more readable 🤔 

```suggestion
    results.try_for_each(future::ok).await
```

Reply:
Neat!